src/Tvheadend.cpp

```diff
@@ -1385,13 +1385,9 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   {
     UPDATE(rec.state, PVR_TIMER_STATE_COMPLETED);
   }
-  else if (strstr(state, "missed") != NULL)
-  {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
-  }
-  else if (strstr(state, "invalid")   != NULL)
+  else
   {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
+    UPDATE(rec.error, state);
```

Comment:
I'm lost. I don't get what exactly do you want t achieve, sorry.

Let's forget for a moment what we have changed so far.

Could you please describe as exact as possible what is buggy with current implementation in master and what exactly do you want change.